source/level_structures.h

```diff
@@ -0,0 +1,197 @@
+/*
+ * level_structures.h
+ *
+ *  Created on: Oct 5, 2023
+ *      Author: jerett
+ */
+
+#pragma once
+
+#include <compile_time_options.h>
+
+//deal.II includes
+#include <deal.II/base/parameter_acceptor.h>
+
+//ryujin includes
+#include "discretization.h"
+#include "hyperbolic_module.h"
+#include "initial_values.h"
+#include "offline_data.h"
+#include "parabolic_module.h"
+#include "postprocessor.h"
+#include "quantities.h"
+#include "time_integrator.h"
+#include "vtu_output.h"
+
+#include <deal.II/base/timer.h>
+#include <deal.II/base/tensor.h>
+
+#include <fstream>
+#include <future>
+#include <sstream>
+
+namespace ryujin{
+  namespace mgrit{
+    ///A struct containing all relevant objects to a given MGRIT level
+    /**
+     * This struct stores all the relevant information for a given level
+     * in the mgrit algorithm.
+     *
+     * Intended usage is as follows:
+     *
+     *    Vector<LevelStructures> levels;
+     *    for(const auto l : mgrit_levels)
+     *      levels.at(l) = LevelStructures(...);
+     *
+     *    //select the relevant level data, eg. offline data
+     *    int level = 0;
+     *
+     *    f(...,levels[level]->offline_data,...);
```

Comment:
I do not understand what `f` is referring to. Do you want to say that this is how you use the offline data of some level as a parameter for some function?

Hmm, maybe just write how you get the offline data, e.g.:
```suggestion
     *    const auto current_offline_data = levels[level]->offline_data;
```